### ISSUE TITLE
Future Waterfox Classic

### ISSUE THREAD

**Issue by roman6626** [MrAlex94/Waterfox#1266]
Does Waterfox Classic have a future? Or is he doomed to extinction? This is essentially a zombie version. Worst of all, he is addicted to Firefox. Because it's not a fork.

**Comment by grahamperrin** [comment]
> Does Waterfox Classic have a future? 

Yes. 

Recent [Waterfox 2019.10 Release | Waterfox Web Browser](https://www.waterfox.net/blog/waterfox-2019.10-release-download/) should have been unmistakably clear: 

>> … There are no plans to retire this branch, and (it) is going to be actively maintained, …

**Comment by roman6626** [comment]
But to be a modern browser, you need to get not only security patches? Not without reason, old versions of browsers stop working in a modern web. The old engine eventually becomes inoperative. New solutions are constantly needed.

**Comment by hawkeye116477** [comment]
@roman6626 Waterfox Classic doesn't only get security patches, if users report that some site isn't working correctly on Classic, then some things are ported.

**Comment by roman6626** [comment]
It's good. Another question: will you do anything in connection with the closure legacycollector? Are you not going to save this archive?

**Comment by grahamperrin** [comment]
The planned shutdown is not a Waterfox issue, please join the discussion at https://redd.it/dtof68

**Comment by roman6626** [comment]
So all the work on Waterfox Classic is just an extension patch? Are there any plans to introduce new features and capabilities into the browser? Why now you can’t work on Firefox 3 or Opera 12? I said about the zombie browser in this aspect. Any browser becomes outdated and stops working normally if it does not support all technologies and changes in a modern web. And security patches alone are not enough.

**Comment by hawkeye116477** [comment]
`security patches alone are not enough`
@roman6626 You're repeating yourself, I already wrote answer => https://github.com/MrAlex94/Waterfox/issues/1266#issuecomment-554742858.

**Comment by roman6626** [comment]
Not. There was no answer. There was a question - Are there any plans to introduce new features and capabilities into the browser?
It turns out that you will correct something only if the user tells you? Is that all? Do you plan to upgrade your browser yourself?

**Comment by grahamperrin** [comment]
> … support all technologies and changes in a modern web. …

Again, please refer to the blog – the post that was linked from my first response here. 

This in particular: 

>> One focused on a legacy branch maintaining security and bug fixes and another branch focusing on keeping up to date with the modern web 

Can I politely ask, is there a language/translation issue?

**Comment by roman6626** [comment]
Not. There are no difficulties with the translation. 
I realized that the Classic branch will only be supported as long as possible and receive security patches. But then she will most likely die, as was the case with the old versions of all browsers that simply stopped working in the modern web.
We are not talking about the Current branch now. It will develop after Firefox.

**Comment by reallyuniquename** [comment]
Think of Waterfox Classic as a Firefox 56 ESR. You get security patches and bug fixes at best. That's it.

A handful of developers provide you with it for free and you should be extremely happy it's happening at all. Now you have a year or two to figure out where to move next and how to recreate legacy extensions and your favorite UI in other modern browsers.

> Are there any plans to introduce new features and capabilities into the browser?

No. You are stuck with Firefox 56 capabilities and performance.

> Does Waterfox Classic have a future?

No. Sooner or later it would not be able to handle modern websites and fixing it would require insane amount of work.

**Comment by roman6626** [comment]
I thought so.
I have a year or two... And then what? And old extensions cannot be recreated or repeated. Modern Firefox is shit. And he will not be better.
And all these projects - Waterfox, Pale Moon, Basilisk, SeaMonkey - these are zombie browsers. They will die because they have no development. They are stuck in time. And very soon they will not be.
Unfortunately.

**Comment by hawkeye116477** [comment]
@roman6626
`They will die because they have no development.`
If you'll look at for example Basilisk changelog => https://www.basilisk-browser.org/releasenotes.shtml, that's not true at all, they took their own path, it's bigger team than in case of Waterfox, so they can do more and faster. Also look at `https://github.com/MoonchildProductions/UXP/issues`, they already have new things in plans.

In case of Waterfox, Classic is legacy branch, so some new things some day may come or not, Current is more promising than Classic. However officially Waterfox team has 1 programmer, so you can't expect much from it. If you want rapid updates, faster following with modern web, then seems this browser isn't for you and you should look for browser which is developed by bigger team like MoonChild or Vivaldi.

**Comment by roman6626** [comment]
Just from experience. 
What does Moonchild do? Forks an obsolete Firefox core and begins to cut out everything "unnecessary" from it. Until it stops working. Then he forks the next Firefox core and again begins to cut out everything "superfluous" from it. Such projects have neither strength, nor knowledge, nor resources.
The standards of the modern web are changing. And you need to modify the browser, engine, functionality, settings, features...

**Comment by roman6626** [comment]
All zombie projects, unfortunately, have 1 (2,3) programmer. And that’s all.

**Comment by roman6626** [comment]
The most important thing in these browsers is the support and work with the old classic extensions. In modern browsers this will never happen again.
Old browsers will stop working after a short time, and all new browsers will be full of crap.
The circle is closed. :-(

**Comment by grahamperrin** [comment]
Is anything specifically actionable here? 

From the opening sentence: 

> Worst of all, he is addicted to Firefox.

The Mozilla code base is not a bad thing. [Mozilla is not an evil empire](https://old.reddit.com/r/waterfox/comments/dtp0q7/today_i_stumbled_upon_just_another_reason_to/f6yutj5/?context=1); and so on. [Maintaining An Independent Browser Is Incredibly Expensive](https://robert.ocallahan.org/2017/12/maintaining-independent-browser-is.html) – I find it disappointing when anyone takes pleasure in encouraging anti-Mozilla sentiment whilst enjoying (with Waterfox Classic and Waterfox Current) the many and varied benefits of Mozilla's work. 

----

Phrases such as *zombie* are unnecessarily harsh, certainly not constructive. 

This issue is neither an enhancement request, nor (heading towards) anything like a pull request …

**Comment by hawkeye116477** [comment]
@roman6626 
`The standards of the modern web are changing.`
Seems that I don't understand you or you aren't looked at their changelog and GH repo, Moochild keeps following with standards of modern web. However they doesn't have that very big team like Google, so they aren't providing a lot of changes in rapid time. But telling that's zombie browser just like in case of Waterfox is exaggerated.

If you want independent browser with own engine, then we don't mind, you can do it and remind us when it's finished and we will express an opinion.

**Comment by roman6626** [comment]
I understand you. The forces, of course, are very unequal. Large corporations and small projects.
And an independent browser alone can not be done. That is why only the assembly.

**Comment by snintendog** [comment]
TLDR what happens when everything turns into google chrome you have to be google chrome in order to even run. Classic is dead netflix is gone youtube elements are failing several sites that are updating their languages complete break even Local servers are breaking waterfox Classic JS errors on everysite breading dropdown menus to animations glitching out. as much as i Love the project the guy is right you are TOO dependent on firefox to the point you are using quantum for the current branch throwing away NPAPI support and not even making the options for the UI NOT to look like Chrome. At this point the project has become less than firefox Removing everything that made it good this project is dead Classic ended support in 56.2, Current is just as big a mess as FF quantum not really fixing anything and not supporting anything.

I hate tabs on top, I hate not having a single Non-unified search engine, I hate the fact i cant send all the status indicators to the bottom of the screen, I hate damn menus that don't do anything but make it a nightmare to navigate to a setting that wont actually do what it says it does, I hate that i cant scroll though my tab history and have to go back or forward 8 to see more, I hate how everything has a EFFING RAM LEAK and I hate using a browser that will lockout customizing it top to bottom, I hate having DRM that will just stop working for no good reason. I HATE CHROME AND EVERYTHING IT STANDS FOR.

**Comment by MrAlex94** [comment]
> netflix is gone

It's not. It doesn't like the UA Waterfox uses. I'll change it with the next update and boom...it works again.

> youtube elements are failing

A UA issue again. It's a challenge finding one that works.

> several sites that are updating their languages complete break

?

>  are TOO dependent on firefox to the point you are using quantum for the current branch

Well, care to offer an alternative? Who is going to pay for the millions of man hours that is required for a modern web engine and the browser that goes around it?

> throwing away NPAPI support

There is NPAPI support in current...

> not even making the options for the UI NOT to look like Chrome

What? You can change the UI like you can on Firefox. There's australis-style and Photon. Then there's the abundant custom CSS that Waterfox allows (and Firefox is soon to drop).

> At this point the project has become less than firefox

Considering there is still support for NPAPI, bootstrap addons, userChrome and userJS, seems like a hell of a lot more customisation options than Firefox?

>not really fixing anything and not supporting anything

This sentence says nothing.

Honestly, everyone here seems to be looking for reasons to be upset. I've provided solutions to what people have been wanting in the best way I can. Volunteers pop in to help. And I have good news to unveil soon as well.

There also seems to be a big lack of technical understand or what's actually happening, and you're coming to rant for no reason. The web is incredibly backwards compatible because the standards are tiered and rely on the previous. Yes, some websites have developers who couldn't give a toss about that and only have  sights for a certain clientele. Nothing much can be done about that. Regardless, things are bleak because you make them so.

Have constructive criticism? Want to have your voice heard? Fine, but the issue reporting system on GitHub is not the place for them, especially when it's so misguided.

**Comment by snintendog** [comment]
3 Things for the entire issue being levied.

1. Sites not to be broken (Current handles this Classic does not) IE Issue #612 this has been happening lately on more sites and it happens on Clean installs Ruling out tampermonkey.  JellyFin is a rather large example of the issue and is replicatable on Classic and other Older version browsers {Firefox 50/IE/Safari} are doing this as well this is when they updated the Web components to the new standards something they and MANY other sites are doing right now. 2nd As you said the UA is an issue but changing the string on my own does register so either netflix does more to checks or something else is stopping 10.2019 classic from spoofing/being accepted. Still tied into the fact these issues are things that will need to be addressed in classic and they will become far more common as time goes on.

2 UI Control without programing knowledge to do it. (Classic supports the CTR addon a workaround Current does not and need to use CSS tweaks) As it is now Most of the most common tweaks such as tabs not on top needs to be done via CSS when a simple addon could do it before now no longer supported at all (despite you saying it should be working just fine).

3. Security patches Both are supporting this.

**Comment by grahamperrin** [comment]
> Honestly, everyone here seems to be looking for reasons to be upset.

+1 

Defocusing from Waterfox Project: I sometimes see vast swathes of the Internet bent out of shape, people impossible to please in communities that are entirely separate from each other, it's like all these disconnected people get bent at the same time with nothing obvious to explain the timing. Like, they're not in the same country (so it can't be the political climate or geographical climate) … I'm simplifying, but there it is. There's get bent then get straight then get bent and so on. I have some theories about this but GitHub is not the place. 

Nor was GitHub the place for this comment, so I'm about to hide it.

**Comment by roman6626** [comment]
Summarize. 
Waterfox Classic will be maintained as long as possible. He will not get anything new in terms of functionality. This is a temporary branch.
Waterfox Current will follow Firefox. And it will be what Waterfox has always been before bifurcation. Old classic extensions will not be supported. It's impossible.
